### PR TITLE
XLS-80d: Permissioned Domains

### PR DESCRIPTION
**Pull request by mvadari**
Discussion thread can be found here: https://github.com/XRPLF/XRPL-Standards/discussions/228

rippled PR: https://github.com/XRPLF/rippled/pull/5149

### CODE REVIEW

**Comment by mvadari** [comment]
cc @mtrippled @oleks-rip for review

File: XLS-0080d-permissioned-domains/README.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+This is an array of inner objects referencing a type of credential. The maximum length of this array is 10.
+
+The array will be sorted by `Issuer`, so that searching it for a match is more performant.
+
+If this field is not included in the domain, then credentials do not need to be used when interacting with anything using the domain (i.e. there are no credential-based rules).
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Issuer`|✔️|`string`|`AccountID`|The issuer of the credential.|
+|`CredentialType`|✔️|`string`|`Blob`|A value to identify the type of credential from the issuer.|
```
Comment (oleks-rip) [inline XLS-0080d-permissioned-domains/README.md:68]: (hex-encoded) as in XLS-70

File: XLS-0080d-permissioned-domains/README.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+|------------|-----------|-----------|---------------|-------------|
+|`TransactionType`|✔️|`string`|`UInt16`|The transaction type (`PermissionedDomainSet`).|
+|`Flags`| |`number`|`UInt32`|Set of bit-flags for this transaction.|
+|`Account`|✔️|`string`|`AccountID`|The account sending the transaction.|
+|`DomainID`| |`string`|`Hash256`|The domain to modify. Must be included if modifying an existing domain.|
+|`AcceptedCredentials`|✔️|`array`|`STArray`|The credentials that are accepted by the domain. Ownership of one of these credentials automatically makes you a member of the domain. An empty array means deleting the field.|
+
+### 3.2. Failure Conditions
+
+* `Issuer` doesn't exist on one or more of the credentials in `AcceptedCredentials`.
```
Comment (oleks-rip) [inline XLS-0080d-permissioned-domains/README.md:90]: CredentialType is empty or longer then 64 bytes

File: XLS-0080d-permissioned-domains/README.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+This value will be used wherever a `DomainID` is required.
+
+#### 2.1.2. `AcceptedCredentials`
+
+This is an array of inner objects referencing a type of credential. The maximum length of this array is 10.
+
+The array will be sorted by `Issuer`, so that searching it for a match is more performant.
+
+If this field is not included in the domain, then credentials do not need to be used when interacting with anything using the domain (i.e. there are no credential-based rules).
```
Comment (Bronek) [inline XLS-0080d-permissioned-domains/README.md:63]: This is confusing to me - the above table shows `AcceptedCredentials` to be required, but the wording here implies that it is optional.

File: XLS-0080d-permissioned-domains/README.md (hunk 1 above)
Comment (Bronek) [inline XLS-0080d-permissioned-domains/README.md:68]: what's the maximum size of this blob ?

File: XLS-0080d-permissioned-domains/README.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+This object represents a permissioned domain.
+
+### 2.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`LedgerIndex`|✔️|`string`|`Hash256`|The unique ID of the ledger object.|
+|`Flags`| ✔️|`number`|`UInt32`|Flag values associated with this object.|
+|`LedgerEntryType`|✔️|`string`|`UInt16`|The ledger object's type (`PermissionedDomain`).|
+|`Owner`|✔️|`string`|`AccountID`|The account that controls the settings of the domain.|
```
Comment (Bronek) [inline XLS-0080d-permissioned-domains/README.md:45]: Should there be also `OwnerNode` ?

File: XLS-0080d-permissioned-domains/README.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+## 3. Transaction: `PermissionedDomainSet`
+
+This transaction creates or modifies a `PermissionedDomain` object.
+
+### 3.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`TransactionType`|✔️|`string`|`UInt16`|The transaction type (`PermissionedDomainSet`).|
+|`Flags`| |`number`|`UInt32`|Set of bit-flags for this transaction.|
```
[the remark this answers is not in the document]
Comment (Bronek) [reply XLS-0080d-permissioned-domains/README.md:83]: NVM I think this meant the common `Flags` field.

File: XLS-0080d-permissioned-domains/README.md (hunk 3 above)
Comment (mvadari) [reply XLS-0080d-permissioned-domains/README.md:63]: The paragraph was left in from when it was optional. Removed.

File: XLS-0080d-permissioned-domains/README.md (hunk 4 above)
Comment (mvadari) [reply XLS-0080d-permissioned-domains/README.md:45]: Good catch, added.

File: XLS-0080d-permissioned-domains/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mvadari) [reply XLS-0080d-permissioned-domains/README.md:83]: It was used in a previous version of the spec. Removed now.

File: XLS-0080d-permissioned-domains/README.md (hunk 1 above)
Comment (mvadari) [reply XLS-0080d-permissioned-domains/README.md]: Added

File: XLS-0080d-permissioned-domains/README.md (hunk 3 above)
Comment (oleks-rip) [reply XLS-0080d-permissioned-domains/README.md]: I believe this is from extended spec when Domain will operate by other rules as well. But for now it is limited by credentials only.

**Pull request merged**